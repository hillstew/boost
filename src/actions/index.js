export const setDogImage = image => ({
  type: 'SET_DOG_IMAGE',
  image
});

export const setCatImage = image => ({
  type: 'SET_CAT_IMAGE',
  image
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setLoading = loading => ({
  type: 'SET_LOADING',
  loading
});

export const removeSaved = id => ({
  type: 'REMOVE_SAVED',
  id
});

export const addSaved = savedBoost => ({
  type: 'ADD_SAVED',
  savedBoost
});
