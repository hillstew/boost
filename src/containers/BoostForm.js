import React, { Component } from 'react';
import { connect } from 'react-redux';
let shortid = require('shortid');
export class BoostForm extends Component {
  constructor() {
    super();
    this.state = {
      senderNum: '',
      recipientNum: '',
      boostMessage: ''
    };
  }

  handleSubmit = e => {
    e.preventDefault();
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSave = e => {
    const { dogImgSrc } = this.props;
    const { senderNum, recipientNum, boostMessage } = this.state;
    let boost = {
      id: shortid.generate(),
      senderNum,
      recipientNum,
      boostMessage,
      dogImgSrc
    };
  };

  render() {
    const { dogImgSrc } = this.props;
    const { senderNum, recipientNum, boostMessage } = this.state;
    const url = 'https://random.dog/' + dogImgSrc;
    return (
      <form onSubmit={this.handleSubmit} className="BoostForm">
        <div className="div-flex">
          <div className="input-div">
            <label>
              To:
              <input
                onChange={this.handleChange}
                name="senderNum"
                type="number"
                placeholder="855-546-1234"
                value={senderNum}
              />
            </label>
            <label>
              From:
              <input
                onChange={this.handleChange}
                name="recipientNum"
                type="number"
                placeholder="765-765-2343"
                value={recipientNum}
              />
            </label>
          </div>
          <img src={url} alt="animal" />
        </div>
        <textarea
          onChange={this.handleChange}
          name="boostMessage"
          placeholder="Write your boost here"
          value={boostMessage}
        />
        <button name="save" onClick={this.handleSave}>
          Save
        </button>
        <button name="send" type="submit">
          Send
        </button>
      </form>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc
});

export default connect(mapStateToProps)(BoostForm);
