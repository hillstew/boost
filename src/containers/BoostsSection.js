import React, { Component } from 'react';
import { BoostCard } from '../components/BoostCard';
import { fetchDogImage } from '../thunks/fetchDogImage';
import { connect } from 'react-redux';

export class BoostsSection extends Component {
  render() {
    const { dogImgSrc, fetchDogImage, isLoading } = this.props;
    return (
      <div>
        {!isLoading && (
          <div className="BoostsSection">
            <h1>Pick your boost image</h1>
            <div className="BoostsSection-cards">
              <BoostCard img={dogImgSrc} fetchDogImage={fetchDogImage} />
            </div>
          </div>
        )}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc,
  isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
  fetchDogImage: () => dispatch(fetchDogImage())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BoostsSection);
