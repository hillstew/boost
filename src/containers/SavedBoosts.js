import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeSaved } from '../actions';

export class SavedBoosts extends Component {
  renderSavedBoosts = () => {
    const { removeSaved, saved } = this.props;
    const url = 'https://random.dog/';
    return saved.map(boost => {
      return (
        <div key={boost.id} className="SavedBoosts-boost">
          <div className="SavedBoosts-flex">
            <div className="div">
              <p>TO: {boost.senderNum}</p>
              <p>FROM: {boost.recipientNum}</p>
            </div>
            <img src={url + boost.dogImgSrc} alt="animal" />
          </div>
          <p className="boost-msg">Your boost messsage: {boost.boostMessage}</p>
          <button name="delete" onClick={() => removeSaved(boost.id)} className="remove-button">
            Delete
          </button>
        </div>
      );
    });
  };

  render() {
    return <div className="SavedBoosts">{this.renderSavedBoosts()}</div>;
  }
}

export const mapStateToProps = state => ({
  saved: state.savedBoosts
});

export const mapDispatchToProps = dispatch => ({
  removeSaved: id => dispatch(removeSaved(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SavedBoosts);
