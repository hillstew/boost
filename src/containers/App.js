import React, { Component } from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { fetchDogImage } from '../thunks/fetchDogImage';
import { connect } from 'react-redux';
import { Nav } from '../components/Nav';
import { Logo } from '../components/Logo';
import { About } from '../components/About';
import { NotFound } from '../components/NotFound';
import BoostsSection from '../containers/BoostsSection';
import BoostForm from '../containers/BoostForm';
class App extends Component {
  componentDidMount() {
    this.props.fetchDogImage();
  }

  render() {
    const { history } = this.props;
    return (
      <div className="App">
        <div className="App-nav-logo-div">
          <Nav />
          <Logo />
        </div>
        <Switch>
          <Route exact path="/" component={BoostsSection} />
          <Route exact path="/about" component={About} />
          <Route
            path="/send-boost/:imgId"
            render={({ match }) => {
              const { imgId } = match.params;
              return <BoostForm img={imgId} history={history} />;
            }}
          />
          <Route path="*" component={NotFound} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  dogImgSrc: state.dogImgSrc,
  isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
  fetchDogImage: () => dispatch(fetchDogImage())
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
