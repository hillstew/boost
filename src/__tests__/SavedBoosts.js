import React from 'react';
import { shallow } from 'enzyme';
import { SavedBoosts } from '../containers/SavedBoosts';
import { removeSaved } from '../actions';
import { mapDispatchToProps } from '../containers/SavedBoosts';
jest.mock('../actions');

describe('SavedBoosts', () => {
  let wrapper;
  let removeSaved;
  let saved;
  let boost;

  beforeEach(() => {
    removeSaved = jest.fn();
    saved = [{}, {}, {}];
    wrapper = shallow(<SavedBoosts removeSaved={removeSaved} saved={saved} />);
    boost = {
      id: '3',
      senderNum: '22',
      recipientNum: '45',
      boostMessage: 'oh hi',
      dogImgSrc: 'hfksjdh-555.jpg'
    };
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call removeSaved when delete is clicked', () => {
    wrapper.find('.delete').simulate('click');
    expect(wrapper.instance().props.removeSaved).toHaveBeenCalledWith(boost.id);
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when removeSaved is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeSaved(boost.id);
      const props = mapDispatchToProps(mockDispatch);
      props.removeSaved(boost.id);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
