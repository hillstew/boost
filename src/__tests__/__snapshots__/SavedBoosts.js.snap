// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedBoosts should match the snapshot 1`] = `
<div
  className="SavedBoosts"
>
  <div
    className="SavedBoosts-boost"
  >
    <div
      className="SavedBoosts-flex"
    >
      <div
        className="div"
      >
        <p>
          TO: 
        </p>
        <p>
          FROM: 
        </p>
      </div>
      <img
        alt="animal"
        src="https://random.dog/undefined"
      />
    </div>
    <p
      className="boost-msg"
    >
      Your boost messsage: 
    </p>
    <button
      className="remove-button"
      name="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="SavedBoosts-boost"
  >
    <div
      className="SavedBoosts-flex"
    >
      <div
        className="div"
      >
        <p>
          TO: 
        </p>
        <p>
          FROM: 
        </p>
      </div>
      <img
        alt="animal"
        src="https://random.dog/undefined"
      />
    </div>
    <p
      className="boost-msg"
    >
      Your boost messsage: 
    </p>
    <button
      className="remove-button"
      name="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="SavedBoosts-boost"
  >
    <div
      className="SavedBoosts-flex"
    >
      <div
        className="div"
      >
        <p>
          TO: 
        </p>
        <p>
          FROM: 
        </p>
      </div>
      <img
        alt="animal"
        src="https://random.dog/undefined"
      />
    </div>
    <p
      className="boost-msg"
    >
      Your boost messsage: 
    </p>
    <button
      className="remove-button"
      name="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
